docs/roadmap.md

```diff
@@ -0,0 +1,169 @@
+# Roadmap
+
+* [**Simple installation**](#simple-installation): importing a project should be
+  simple and require as few steps as possible.
+* [**Correct diagnostics**](#correct-diagnostics): the editor should never show
+  spurious red squiggles, even at the cost of higher latency.
+* [**Robust navigation**](#robust-navigation): goto definition should work
+  everywhere, even for Java dependencies.
+* [**Fast completions**](#fast-completions): completions suggestions should
+  respond in ~200ms, even at the cost of occasionally wrong results.
+* [**Low CPU and memory usage**](#low-cpu-and-memory-usage): indexing should be
+  lightweight and run in the background.
+* [**Solid refactoring**](#solid-refactoring): rename, organize imports and
+  insert type annotation should always leave the codebase in a compiling state.
+
+## Simple installation
+
+The first you do with an IDE is boring but critical: import a project. The
+Language Server Protocol does not provide utilities to extract metadata from a
+build tool (module source files, dependencies, configuration data), so we are
+still exploring what is the best option.
+
+* [ ] sbt-metals: plugin that provides a `*:metalsSetup` task to export sbt
+      build metadata.
+* [ ] [BSP][]: LSP-inspired protocol to standardize on communication between a
+      language server and build tool. In theory, should enable automatic
+      importing of projects without custom build plugins or manual installation
+      steps.
+
+## Correct diagnostics
+
+Red squiggles are one of the most basic but also difficult-to-support features
+in an IDE. The Scala compiler was not designed for interactivity, which makes
+fast and correct diagnostics difficult to support in the editor. We are still
+unsure what the best model is for diagnostics, so we are still exploring what is
+the best option.
+
+* [x] Scala Presentation Compiler: interactively publish squiggles as you type.
+      Provides low latency but may get off-sync with build diagnostics.
+  * Disabled by default. Can be enabled with configuration option
+    `scalac.diagnostics.enabled=true`.
+  * Requires [`metalsSetup`](#metalssetup).
+* [x] sbt server: sbt 1.1 added LSP support so that build errors are reported
+      directly to the editor. Metals can connect to sbt server as a client to
+      trigger batch compilation on file save. Guarantees diagnostics are always
+      in-sync with build diagnostics but at the cost of higher latency.
+  * Disabled by default. Can be enabled with configuration option
+    `sbt.enabled=true`.
```

Comment:
Why is it `scalac.diagnostics.enabled`, but not `sbt.diagnostics.enabled`?

Reply:
it should probably be `sbt.diagnostics.enabled`, but that's for another PR

Reply:
tracked in https://github.com/scalameta/metals/issues/228

Reply:
If it's enabled by default, then it shouldn't show an error message when it fails to connect _automatically_ (only when user calls the "sbt connect" command). Imagine you use it for the first time and you don't know anything about this integration and the first thing you see after  server initialization is an error: "sbt 1.1 is not running..." ("wtf, I'm not even using sbt" says somebody with bloop/mill/etc. 😝)